Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/checkins/CheckInsTransformer.kt

```diff
@@ -61,16 +62,38 @@ class CheckInsTransformer @Inject constructor(
                     checkInStart = derivedTimes.startTimeSeconds.secondsToInstant(),
                     checkInEnd = derivedTimes.endTimeSeconds.secondsToInstant()
                 )
-                derivedCheckIn.splitByMidnightUTC().map { checkIn ->
+                derivedCheckIn.splitByMidnightUTC().mapNotNull { checkIn ->
                     checkIn.toOuterCheckIn(transmissionVector)
                 }
             }
         }
     }
 
-    private fun CheckIn.toOuterCheckIn(
+    private suspend fun CheckIn.toOuterCheckIn(
         transmissionVector: TransmissionRiskVector
-    ): CheckInOuterClass.CheckIn {
+    ): CheckInOuterClass.CheckIn? {
+        val trvMappings: List<TransmissionRiskValueMapping> =
+            appConfigProvider.getAppConfig()
```

Comment:
For performance reasons, and to prevent the config from changing during calculation, please re-use the retrieved one from line 42.